### DISCUSSION TITLE
Jacobian caculation formulas

### DISCUSSION THREAD

**Discussion by GhiathMonnet** [thelfer/tfel#544] [General]
Thank you Thomas, I have a last question :
I saw in the document how the consistent tangent is computed and I understand it now.  
I am working with @DSL Implicit and declared one state variable p for the equivalent strain.
For the calculation of the Jacobian, I beleive I have to provide : 
 - dfeel_ddeel 
 - dfeel_ddp  
 - dfp_ddp
 - dfp_ddeel
I recongnised that dfeel_ddp = n, and saw that you described how to express dfp_ddp in the document "behavior". But couldn't see a trivial derivation of the other two derivatives (dfeel_ddeel , dfp_ddeel). Did you describe somewhere how to obtain theme (as you have done for the consistente tangent opertaor) ? Or may be they are described in other documents ?
Cheers.
Ghiath

**Comment by thelfer** [comment]
Those derivatives depends on the consistutive law treated. Here is a detailed example: https://thelfer.github.io/tfel/web/isotropicplasticityamstrongfrederickinematichardening.html

**Reply by GhiathMonnet** [reply]
OK I see, thanks.
Now, in my case (one stat variable p), it is difficult for me to understand exactly dfeel_ddeel with respect to dfeel_ddp.
In my case deel is always equal to deto - dp *n. Since deto is constant over the iteration, p and deel are not independent variables. To keep consistency of the deformation decomposition, dp cannot vary without a variation of deel. What is the meaning of these partial derivatives with respect to dependent variables ? May be I misunderstood again :)

**Reply by thelfer** [reply]
I don't get your point. `deel` and `dp` are the solutions of an implicit system at fixed `deto`.
When solving the implicit system, they are considered as independent variables, i.e. they can have (almost) arbitrary values up to convergence of the Newton method.
For the computation of the consistent tangent operators, we consider the solutions of the implicit system which are indeed implicit functions of `deto`.

**Comment by GhiathMonnet** [comment]
OK I see. 
Just to understand more precisely, since  deel is always equal to deto - dp *n. Can we instead of using @DSL Implicit (two state variables eel and p), we use the @DSL ImplicitII (which does not declare eel) and : 
- write the implicit scheme to compute dp  ;
- update the stress by sig = D * (eel_old + deto - p * n), where eel_old is an @AuxiliaryStateVariable saving the elastic strain  ;
- Such implementation is much simpler than the @Implicit one. Does such implementation converge faster or slower ?

**Reply by thelfer** [reply]
You are partially right, this second scheme is valid.

But if you explicitly write `deel = deto - dp *n`, then you change your implicit scheme: you don't solve the same system of nonlinear equations.
